frontend/vite-project/src/components/payment-form/PaymentForm.tsx

```diff
@@ -0,0 +1,179 @@
+import { useState } from "react";
+import { usePaymentStore } from "../../stores/paymentStore";
+import { ValidationRule } from "../../types/paymentFormTypes";
+import swishIcon from '../../assets/image/swish.svg';
+import klarnaIcon from '../../assets/image/klarna.svg';
+import "../payment-form/PaymentForm.css";
+
+const PaymentForm = () => {
+    const { formData, updateFormField } = usePaymentStore();
```

Comment:
```suggestion
   const formData= useFormData();
   const {updateFormField} = usePaymentActions();
```
I'm sure you did this before I realized my mistake. Leaving it as it was would probably be fine for now, but if we added another piece of state, it could lead to issues. I just want to encourage good habits by using custom hooks for accessing state.